### ISSUE TITLE
Failed to subscribe to new newBlockHeaders to confirm the transaction receipts.

### ISSUE THREAD

**Issue by damarnez** [walletlink/walletlink#19]
When I try to interact with my smart contracts and I wait the response with a promise or async , the conection web3 connected to the wallet walletlink throw this error:

![image](https://user-images.githubusercontent.com/1387222/69644010-5cec8580-1064-11ea-911e-628f25383d62.png)

I write a simple example, for test and appears the same error:
https://github.com/damarnez/walletlink_example

The line that we get the throw are :

`await contracts.TestDani.methods.set(magicNumber).send({
      from: address 
 });`

Is a simple transaction and wait the response from the chain. With  others wallets works perfectly..

Feels like the problem is Infura doesn't allow newBlockHeaders by spec.https://infura.io/docs/#supported-json-rpc-methods

What can I do? 

Infura :   https://kovan.infura.io/v3/ ****
Web3 version : 1.2.1
WalletLink: 1.0.0

**Comment by swkim109** [comment]
Same issue here. I have tried to use `wss://` but did not work.
Infura: https://rinkeby.infura.io/v3/***
Web3 1.2.4
Walletlink 1.0.0
And is there any websocket provider such as `walletLink.makeWebsocketProvider`?

**Comment by mrjoedang** [comment]
Same issue here (using the Kovan network of Infura) when trying to authorize a transaction.

**Comment by damarnez** [comment]
I was check the source and the problem is the walletlink provider doesn't support websocket. One fast solution is make this fix on your source.

```
  const walletLink = new WalletLink({
    appName: 'Example Company',
    appLogoUrl: `https://static.example.com/favicons/favicon.ico`
  });

  const wallet = walletLink.makeWeb3Provider(
    `https://mainnet.infura.io/v3/YOUR_TOKEN_INFURA`,
    1
  );

// FIX BUG: Walletlink bug with web3
wallet.on = null;

```
With this fix, we force the Provider to conect to the chaind only with https.

**Comment by petejkim** [comment]
[This is fixed in 2.0.0](This is fixed in 2.0.0)

**Comment by petejkim** [comment]
[This is fixed in 2.0.0](https://www.npmjs.com/package/walletlink/v/2.0.0-rc.1)

**Comment by damarnez** [comment]
Thanks!! :D
